src/components/LocaleLink.js

```diff
@@ -21,8 +21,8 @@ const LocaleLink = ({ children, to, locale, ...other }) =>
       }
 
       localeTo = localeTo.replace(
-        new RegExp(`^/${supportedLocales.join('/|')}(.*)$`),
-        `${locale === defaultLocale ? '' : locale + '/'}$1`
+        new RegExp(`^//${supportedLocales.join('/|/')}/(.*)$`),
```

Comment:
Because regexps are bananas, I think the very first slash is already covered:

```
> const supportedLocales = ["en","es","de","ru"]
< undefined
> `^//${supportedLocales.join('/|/')}/(.*)$`
< "^//en/|/es/|/de/|/ru/(.*)$"
```

In fact, I wonder if this is `^/${supportedLocales.join('|')/`, and we drop the `$1` from the `replace` call? That is, we look specifically for a locale between `/`, and specifically replace that with the result below. Then the replacement becomes `locale === defaultLocale ? '' : locale`?

I haven't dived all the way into how the locale stuff is done, so I might be way off.

Reply:
> Because regexps are bananas

🙃:joy:😭 

I'll test it out!

Reply:
Took out the extra backslash! Had to keep the search to include the backslashes instead of just the locale in between it because there is an edge case where the locale just gets dropped and you end up with a double backslash `//my-foo-bar`...I tried avoiding that by only executing replacement if the locale didn't match the default locale, but then links that include an extraneous `en` like `/en/foo-bar` aren't reformatted to `/my-bar`.

Reply:
Ell. Oh. Ell.